vignettes/fims-path-maturity.Rmd

```diff
@@ -0,0 +1,614 @@
+---
+title: "FIMS Path: Maturity"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{FIMS Path: Maturity}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+
+```{css, echo=FALSE}
+
+.rchunk {
+  border: solid #2e75b6;
+}
+.rcppchunk {
+  border: solid #c55a11;
+}
+.fimschunk {
+  border: solid #548235;
+}
+.infochunk {
+  border: solid #262626;
+}
+```
+
+
+<style type="text/css">
+code span.co{
+  color: #8c8c8c; 
+}
+</style>
+
+This vignette describes the hierarchical structure of FIMS by describing the linkages, or *path* through the FIMS model using maturity as an example. The vignette is tailored to FIMS developers or others interested in understanding the FIMS core architecture. The vignette walks through how input from R is carried into the core FIMS calculations. 
+
+The vignette will build up components to the final path:
+```{r image-path-maturity-8a, echo = FALSE, message=FALSE, out.width = '100%'}
+knitr::include_graphics("figures/fims-path-maturity-8.png")
+```
+
+Code chunks in this vignette will be color coded similar to this diagram: R code in blue, Rcpp interface code in orange, fims core code in green, and information in grey.
+
+## Setting up a module in R
+
+We'll start from R. The first step is to install FIMS, retrieve the FIMS module, create a new maturity module inside FIMS, and populate the maturity module with parameter values. 
+
+```{r, class.source = "rchunk", eval = FALSE}
+# install and load FIMS
+remotes::install_github("NOAA-FIMS/FIMS@integration-test-estimation-R-clear-FIMS")
+library(FIMS)
+# Create a new maturity model
+fims <- Rcpp::Module("fims", PACKAGE = "FIMS")
+maturity <- new(fims$LogisticMaturity)
+# Populate the maturity module with parameter values.
+maturity$median$value <- 10
```

Comment:
thought: in the future it might be good to not name parameters the same name as major functions in R.

Reply:
Created issue #412